### PR TITLE
fix links in KubeOne docs

### PR DESCRIPTION
**Pull request by dakraus**
Signed-off-by: Daniel Kraus <daniel.kraus@kubermatic.com>

### CODE REVIEW

File: content/kubeone/main/architecture/requirements/infrastructure-management/_index.en.md [hunk 1]
```diff
@@ -161,9 +161,9 @@ The needed `output.tf` file already comes with all our
 
 [kubeadm-sysreq]: https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/install-kubeadm/#before-you-begin
 [ssh]: {{< ref "../../../guides/ssh#sshd-requirements-on-instances" >}}
-[supported-providers]: {{< ref "../../compatibility#supported-providers" >}}
-[supported-os]: {{< ref "../../compatibility#supported-operating-systems" >}}
-[terraform-configs-github]: https://github.com/kubermatic/kubeone/tree/release/v1.3/examples/terraform
+[supported-providers]: {{< ref "../../supported-providers/" >}}
+[supported-os]: {{< ref "../../compatibility/os-support-matrix/" >}}
+[terraform-configs-github]: https://github.com/kubermatic/kubeone/tree/master/examples/terraform
```
Comment (xmudrii) [inline content/kubeone/main/architecture/requirements/infrastructure-management/_index.en.md:166]:
```suggestion
[terraform-configs-github]: https://github.com/kubermatic/kubeone/tree/main/examples/terraform
```

File: content/kubeone/v1.5/architecture/requirements/infrastructure-management/_index.en.md [hunk 2]
```diff
@@ -161,9 +161,9 @@ The needed `output.tf` file already comes with all our
 
 [kubeadm-sysreq]: https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/install-kubeadm/#before-you-begin
 [ssh]: {{< ref "../../../guides/ssh#sshd-requirements-on-instances" >}}
-[supported-providers]: {{< ref "../../compatibility#supported-providers" >}}
-[supported-os]: {{< ref "../../compatibility#supported-operating-systems" >}}
-[terraform-configs-github]: https://github.com/kubermatic/kubeone/tree/release/v1.3/examples/terraform
+[supported-providers]: {{< ref "../../supported-providers/" >}}
+[supported-os]: {{< ref "../../compatibility/os-support-matrix/" >}}
+[terraform-configs-github]: https://github.com/kubermatic/kubeone/tree/master/examples/terraform
```
Comment (xmudrii) [inline content/kubeone/v1.5/architecture/requirements/infrastructure-management/_index.en.md:166]: Docs for stable releases should point to the release branches.

```suggestion
[terraform-configs-github]: https://github.com/kubermatic/kubeone/tree/release/v1.5/examples/terraform
```

**Comment by dakraus** [comment]
@xmudrii thanks for the suggestions. Both changes are included in the latest commit.

**Review by xmudrii — APPROVED** [review]
/lgtm
/approve

**Pull request merged**